### DISCUSSION TITLE
Can anyone help me connect the dots on my understanding about fuzz and fuzzmore logic used in smm_ptr module(Pg 81 of 99) intel chipsec-manual.

### DISCUSSION THREAD

**Discussion by girishd22** [chipsec/chipsec#1575] [Q&A]
Context: smm_ptr module(Page 81 of 99)

There are 2 modes fuzz and fuzzmore. Can one of you brief how these 2 modes work?

In the eagles view, I understand that below mentioned actions are happening
1. Allocate memory and fill with pattern.
2. Set registers to address of allocated memory.
3. Invoke SW SMI.
4. Check fill pattern.

Will triggering the SMI impact the data stored in the memory buffer and will this in turn result in unchecked input pointers ?

Thanks,
Girish

**Comment by npmitche** [comment]
I believe, `fuzz` will create a pointer and fill it in with a know value to see if the pointer value changes.  Where as `fuzzmore` will create 2 pointers where the first points to the second to see if the memory in the second changes.  Essentially they are trying to mimic if the value passed in is treated directly as a data or a pointer and then ensuring that the SMM doesn't modify it's data.
